### PR TITLE
FIX: Validate format of links

### PR DESCRIPTION
**Pull request by jbrw**
Attempt to validate the format of URLs for image and video attributes, and remove any values that appear to be invalid.

### CODE REVIEW

File: lib/onebox/helpers.rb [hunk 1]
```diff
@@ -178,6 +178,10 @@ def self.normalize_url_for_output(url)
       url.gsub!("'", "&apos;")
       url.gsub!('"', "&quot;")
       url.gsub!(/[^\w\-`.~:\/?#\[\]@!$&'\(\)*+,;=%\p{M}’]/, "")
+
+      parsed = Addressable::URI.parse(url)
+      return "" unless parsed.host && parsed
```
Comment (CvX) [inline lib/onebox/helpers.rb:183]:
```suggestion
      return "" unless parsed.host
```

Are there any cases where `parsed` could be falsy? Other than `url` being nil, but that won't happen in this case.

File: lib/onebox/open_graph.rb [hunk 2]
```diff
@@ -32,7 +32,8 @@ def method_missing(attr, *args, &block)
       if method_name.end_with?(*integer_suffixes)
         value.to_i
       elsif method_name.end_with?(*url_suffixes)
-        ::Onebox::Helpers.normalize_url_for_output(value)
+        result = ::Onebox::Helpers.normalize_url_for_output(value)
+        Onebox::Helpers::blank?(result) ? nil : result
```
Comment (CvX) [inline lib/onebox/open_graph.rb:36]: `::Onebox` or `Onebox`? 😃 

Also, maybe: `result unless Onebox::Helpers::blank?(result)`

**Review by CvX — CHANGES_REQUESTED** [review]

**Comment by CvX** [comment]
@jbrw ping 😃

**Comment by jbrw** [comment]
> @jbrw ping 😃

Oh - this totally fell off my radar. Sorry about that.

I have made a note to self and will circle back to this tomorrow.

**Review by CvX — APPROVED** [review]

**Pull request merged**